core/src/View/MDSpan/Kokkos_MDSpan_Layout.hpp

```diff
@@ -77,45 +81,106 @@ KOKKOS_INLINE_FUNCTION auto array_layout_from_mapping(
         rank > 7 ? mapping.stride(7) : 0,
     };
   } else {
-    // FIXME: Kokkos Layouts don't store stride (it's in the mapping)
-    // We could conceivably fix this by adding an extra ViewCtorProp for
-    // an abritrary padding. For now we will check for this.
-    if constexpr (rank > 1 &&
-                  (std::is_same_v<typename mapping_type::layout_type,
-                                  Kokkos::Experimental::layout_left_padded<
-                                      dynamic_extent>> ||
-                   std::is_same_v<typename mapping_type::layout_type,
-                                  Kokkos::Experimental::layout_right_padded<
-                                      dynamic_extent>>)) {
-      [[maybe_unused]] constexpr size_t strided_index =
-          std::is_same_v<
-              typename mapping_type::layout_type,
-              Kokkos::Experimental::layout_left_padded<dynamic_extent>>
-              ? 1
-              : rank - 2;
-      [[maybe_unused]] constexpr size_t extent_index =
-          std::is_same_v<
-              typename mapping_type::layout_type,
-              Kokkos::Experimental::layout_left_padded<dynamic_extent>>
-              ? 0
-              : rank - 1;
-      KOKKOS_ASSERT(mapping.stride(strided_index) == ext.extent(extent_index));
-    }
-
-    return ArrayLayout{rank > 0 ? ext.extent(0) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
+    ArrayLayout layout{rank > 0 ? ext.extent(0) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                        rank > 1 ? ext.extent(1) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                        rank > 2 ? ext.extent(2) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                        rank > 3 ? ext.extent(3) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                        rank > 4 ? ext.extent(4) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                        rank > 5 ? ext.extent(5) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                        rank > 6 ? ext.extent(6) : KOKKOS_IMPL_CTOR_DEFAULT_ARG,
                        rank > 7 ? ext.extent(7) : KOKKOS_IMPL_CTOR_DEFAULT_ARG};
+
+    if constexpr (rank > 1 &&
+                  std::is_same_v<typename mapping_type::layout_type,
+                                 Kokkos::Experimental::layout_left_padded<
+                                     dynamic_extent>>) {
+      layout.stride = mapping.stride(1);
+    }
+    if constexpr (std::is_same_v<typename mapping_type::layout_type,
+                                 Kokkos::Experimental::layout_right_padded<
+                                     dynamic_extent>>) {
+      if constexpr (rank == 2) {
+        layout.stride = mapping.stride(0);
+      }
+      if constexpr (rank > 2) {
+        if (mapping.stride(rank - 2) != mapping.extents().extent(rank - 1))
```

Comment:
This is from LayoutRIght having a left padding (unlike layout_right_padded) right?

Reply:
yes, and no we won't make a custom LayoutRight, we just gonna disable padding for Rank>3. As I said above we never leverage the stride capability (e.g. in subview). So we are actually fine.